### PR TITLE
log errno names instead of values

### PR DESCRIPTION
**Pull request by udit043**
https://project.freertc.org/issues/25 

In Order to solve this bug, I have created one header file **[Errdes.hxx](https://github.com/udit043/resiprocate/blob/master/rutil/Errdes.hxx)** inside _rutil_ directory. It has error messages associated with the error number/code from the following functions :

-> [SSL_ERROR_SYSCALL](http://www.cl.cam.ac.uk/cgi-bin/manpage?3+errno) 
-> [SSL_get_error](https://www.openssl.org/docs/manmaster/ssl/SSL_get_error.html)
-> [X509_verify_cert_error_string](https://www.openssl.org/docs/manmaster/crypto/X509_STORE_CTX_get_error.html)

I have updated all the files(32) that contains numeric log messages. Now these files will call function from [Errdes.hxx](https://github.com/udit043/resiprocate/blob/master/rutil/Errdes.hxx) and will log error message instead of error code.

Example:
int e = getErrno();
ErrLog( << "socket error " << e);
At present, logs a numeric error code, e.g.
  "socket error 22"

After [commit](https://github.com/resiprocate/resiprocate/pull/47/files#diff-31d616efd5b81206aa0638b4b745dc20L268), 
ErrLog( << "socket error " << errortostringOS(e) );
It will log error message, e.g.
"socket error EINVAL (Invalid argument) 22"
If error number is negative or unknown error occur then log message will be,
"socket error Unknown error errno" 
(e.g. "socket error Unknown error -48")

Total 34 files changed: 32-files updated + 1-header file created + configure.ac file updated for successful build in Travis-CI
 
For quick view of all the files that i have updated please [visit](http://paste.debian.net/hidden/11d413da/)

### CODE REVIEW

**Comment by dpocock** [comment]
@dbozzali @sgodin could either of you comment on Errdes.hxx added in this commit?  Does it look good for Windows?

File: rutil/Errdes.hxx [hunk 1]
```diff
@@ -0,0 +1,268 @@
...
+/*EOWNERDEAD      130 */  "EOWNERDEAD (Owner died)",
+/*ENOTRECOVERABLE 131 */  "ENOTRECOVERABLE (State not recoverable)",
+/*ERFKILL         132 */  "ERFKILL (Operation not possible due to RF-kill)",
+/*EHWPOISON       133 */  "EHWPOISON (Memory page has hardware error)",
+};
+static const int NUM_MESSAGES_OS = sizeof(messagesOS)/sizeof(messagesOS[0]);
+
+static inline char* errortostringOS(int errnumOS)
+{
+   if (errnumOS < NUM_MESSAGES_OS)
```
Comment (dpocock) [inline rutil/Errdes.hxx]: what if errnumOS is a negative value?  Should the function expected an unsigned value or should the range check ensure it is >= 0?

File: rutil/Errdes.hxx [hunk 2]
```diff
@@ -0,0 +1,268 @@
...
+      return messagesOS[errnumOS];
+   return "Unknown error";
```
Comment (dpocock) [inline rutil/Errdes.hxx]: please add the value of errnumOS to this string

**Comment by dpocock** [comment]
@udit043 - please ensure any new files you add (such ass Errdes.hxx) have a license text, you can copy the text from the bottom of https://github.com/resiprocate/resiprocate/blob/master/repro/plugins/pyroute/PyRoutePlugin.cxx

Note that many files in the reSIProcate project have the old VOCAL license, do not use that license in any new files you create.

File: rutil/Errdes.hxx (hunk 1 above)
Comment (udit043) [inline rutil/Errdes.hxx]: I made some changes to code 
```
static inline char* errortostringOS(int errnumOS)
{
   if ( (errnumOS >= 0 )  &&  (errnumOS < NUM_MESSAGES_OS))
      return messagesOS[errnumOS];
   return "Unknown error";
}
```
It is not possible for errnumOS to be negative, please read these two answers
[answer1](http://stackoverflow.com/a/25489728/4499919) [answer2](http://stackoverflow.com/a/18351359/4499919)

File: rutil/Errdes.hxx (hunk 2 above)
Comment (udit043) [inline rutil/Errdes.hxx]: Okay, working on it.

**Comment by udit043** [comment]
I accidentally execute one command on git and my repository goes back into history, I reset it using 
`git reset --hard HEAD@{0}` and in between this my Pull Request automatically gets closed.

**Pull request closed without merging**